src/server/src/main/java/io/cassandrareaper/resources/CryptoResource.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2014-2017 Spotify AB
+ * Copyright 2016-2018 The Last Pickle Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.cassandrareaper.resources;
+
+import io.cassandrareaper.AppContext;
+import io.cassandrareaper.crypto.Cryptograph;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/crypto")
+@Produces(MediaType.TEXT_PLAIN)
+public final class CryptoResource {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CryptoResource.class);
+
+  private final Cryptograph cryptograph;
+
+  public CryptoResource(AppContext context) {
+    this.cryptograph = context.cryptograph;
+  }
+
+  @GET
+  @Path("/encrypt/{text}")
+  public Response encrypt(@PathParam("text") String text) {
+    try {
+      return Response.ok().entity(cryptograph.encrypt(text)).build();
+    } catch (RuntimeException e) {
```

Comment:
question: can this catch be more specific? and the response more specific?

Reply:
Like a ReaperException?  Or are you thinking more along the lines of a new exception type like: CipherException or something?

Reply:
I was thinking of keeping it unchecked. But come to think about it doesn't the codebase catch have handle runtime exceptions already for us? (ie you don't need a catch here at all…?)

Reply:
oh that is interesting.  i didn't realize that dropwizard had a default Exception mapper.  Yep, I will definitely remove this try/catch.